projects/exui-common-lib/src/lib/components/write-address/write-address.component.ts

```diff
@@ -0,0 +1,158 @@
+import { Component, EventEmitter, Input, OnChanges, OnInit, Output, SimpleChanges } from '@angular/core';
+import { FormControl, FormGroup } from '@angular/forms';
+import { AddressModel } from '../../models';
+import { AddressOption } from '../../models/address-option.model';
+import { AddressService } from '../../services/address/address.service';
+
+@Component({
+  selector: 'xuilib-write-address-field',
+  styleUrls: ['./write-address.component.scss'],
+  templateUrl: './write-address.component.html'
+})
+export class WriteAddressFieldComponent implements OnInit, OnChanges {
+
+  public addressesService: AddressService;
+
+  public addressField: AddressModel = {
+    addressLine1: '',
+    addressLine2: '',
+    addressLine3: '',
+    postCode: '',
+    postTown: '',
+    country: '',
+    county: ''
+  };
+
+  @Input()
+  public formGroup: FormGroup;
+
+  @Input()
+  public isExpanded = false;
+
+  @Input()
+  public internationalMode = false;
+
+  @Output() public internationalModeStart = new EventEmitter<void>();
+
+  public isInternational: boolean;
+  public ukRadioChecked = false;
+  public addressChosen = false;
+
+  public addressFormGroup = new FormGroup({});
+  public ukInternationalFormGroup = new FormGroup({});
+  public postcode: FormControl;
+  public addressList: FormControl;
+  public ukAddress: FormControl;
+
+  public addressOptions: AddressOption[];
+
+  public missingPostcode = false;
+
+  constructor(addressesService: AddressService) {
+    this.addressesService = addressesService;
+  }
+
+  public ngOnInit(): void {
+    if (!this.formGroup.get('address')) {
+      this.formGroup.addControl('address', new FormControl({}));
+    }
+    this.ukInternationalFormGroup = new FormGroup({
+      ukAddress: new FormControl()
+    });
+    this.postcode = new FormControl('');
+    this.addressFormGroup.addControl('postcode', this.postcode);
+    this.addressList = new FormControl('');
+    this.addressFormGroup.addControl('address', this.addressList);
+    this.ukAddress = new FormControl('');
+    this.addressFormGroup.addControl('ukAddress', this.ukAddress);
+  }
+
+  public findAddress() {
```

Comment:
@johnbenjamin-hmcts - missing return type

Reply:
@Daniel-Lam - I have updated the code.